### PR TITLE
Bump minimum Python to 3.10

### LINKED ISSUE #28489: [TST] Upcoming dependency test failures
The weekly build with nightly wheels from numpy and pandas
has failed. Check the logs for any updates that need to be
made in matplotlib.
https://github.com/matplotlib/matplotlib/actions/runs/9721989654

### PR DESCRIPTION
**Pull request by QuLogic**
## PR summary

It's time for this by SPEC0, and our nightly build has started failing since Pandas has already dropped 3.9 support.

Fixes #28489

Draft because I need to see if I got the minimum Qt dependencies right.

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by QuLogic** [comment]
It looks like we have problems building on PyPy3.10 because NumPy doesn't have wheels yet https://github.com/numpy/numpy/issues/26141

Also, not sure if the original test condition on the ipynb test is wrong, since it's failing here.

**Comment by tacaswell** [comment]
I think we just skip pypy310 on at least windows, maybe all platforms.

**Comment by QuLogic** [comment]
I added a `ipykernel==5.5.6` requirement to the older ipython job, as pip was walking back the versions to find a compatible one, so probably best to help it along with a pin.

However, `test_ipynb` is failing, and it seems like we are getting `module://ipykernel.pylab.backend_inline` instead of our expected `module://ipykernel.pylab.backend_inline`, and I'm not sure why it's only happening now with Python 3.10 instead of the older 3.9, but I can reproduce locally. cc @ianthomas23

**Comment by ianthomas23** [comment]
I think your ipython pin should be 7.29.0 as that was released in October 2021, the same month as Python 3.10. That is consistent with ipykernel 5.5.6. The test passes locally for me using these.

**Review by tacaswell — APPROVED** [review]

**Comment by QuLogic** [comment]
I forgot about Cygwin here; I believe they only have 3.9, and the plan is to jump to 3.12 directly: https://github.com/numpy/numpy/pull/26222#issuecomment-2047883422

We'll have to disable it in the meantime.

**Comment by rcomer** [comment]
Could we also add a minimum pin for python in [environment.yml](https://github.com/matplotlib/matplotlib/blob/main/environment.yml)?  For some reason mamba gives me 3.9 by default, but it also resolves fine if I pin it higher.

File: .github/workflows/cygwin.yml [hunk 1]
```diff
@@ -49,10 +49,12 @@ jobs:
   test-cygwin:
     runs-on: windows-latest
     name: Python 3.${{ matrix.python-minor-version }} on Cygwin
+    # Enable these when Cygwin has Python 3.12.
     if: >-
       github.event_name == 'workflow_dispatch' ||
-      github.event_name == 'schedule' ||
+      (false && github.event_name == 'schedule') ||
```
Comment (ksunden) [inline .github/workflows/cygwin.yml:55]: Did you mean to keep the `false` here?

Comment (ksunden) [reply .github/workflows/cygwin.yml:55]: answer: yes, cygwin is skipping 3.10 and 3.11, but is not ready for 3.12, so will be re-enabled later

**Review by ksunden — APPROVED** [review]

**Pull request merged**